webui/react/src/utils/data.ts

```diff
@@ -1,7 +1,7 @@
-import { CommandMisc, CommandState, MetricName, MetricType, RunState, State } from 'types';
+import { CommandMisc, CommandState, MetricName, MetricType, RawJson, RunState, State } from 'types';
 
 export const isMap = <T>(data: T): boolean => data instanceof Map;
-export const isNumber = <T>(data: T): boolean => typeof data === 'number';
+export const isNumber = (data: unknown): data is number => typeof data === 'number';
```

Comment:
we should probably change the rest to unknown as well so we can set the return type like this then typescript understands we're guarding types.

Reply:
unless we are actually setting these generic types. thoughts?

Reply:
yeah unknown + X is type is the correct way to go

Reply:
totally agree, if you want to update them in this PR that's great. Or if you want to leave it for a separate PR that's fine too. If you do the later let me know and I'll create the ticket for it.

Reply:
I'd wanna check their references to see if the type guard would let us make their return values safer in places these are used (like it did with isNumber). we shouldn't delay this for that

Reply:
https://determinedai.atlassian.net/browse/DET-3938